Shopify/shopify-app-js
# Authenticate Fulfillment Service Requests

### WHY are these changes introduced?

Fixes #578 


### WHAT is this pull request doing?
* Adds new API to authenticate fulfillment service notifications
* [For fulfillment management apps](https://shopify.dev/docs/apps/fulfillment/fulfillment-service-apps/manage-fulfillments)

#### To test
* Add a new route in your app to receive fulfillment notifications `<callback_url>/fulfillment_order_notification`
* Create a fulfillment service
```
mutation fulfilmentServiceCreate {
    fulfillmentServiceCreate(name: "my valid fulfillment service",
    fulfillmentOrdersOptIn: true,
    callbackUrl: "https://my-tunnel.trycloudflare.com",) {
      fulfillmentService {
        handle
        id
        callbackUrl
      }
    userErrors {
      field
      message
    }
    }
  }
```

* In Shopify Admin update the location of a product to the new fulfillment service
* Create a draft order with the product
* Click `Request fulfillment`
* Verify your server logs that requests are being authenticated

<img width="1517" alt="image" src="https://github.com/Shopify/shopify-app-js/assets/23265671/5b4d3f8b-c5bb-4496-963d-c955c8f54f65">


## Type of change

- [ ] Patch: Bug (non-breaking change which fixes an issue)
- [x] Minor: New feature (non-breaking change which adds functionality)
- [ ] Major: Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [x] I have used `yarn changeset` to create a draft changelog entry (do NOT update the `CHANGELOG.md` files manually)
- [x] I have documented new APIs/updated the documentation for modified APIs (for public APIs)
